plugin/storage/es/spanstore/dbmodel/to_domain.go

```diff
@@ -126,10 +134,40 @@ func (td toDomain) convertKeyValues(tags []json.KeyValue) ([]model.KeyValue, err
 	return retMe, nil
 }
 
+func addTagsMap(tagsMap map[string]interface{}, deDotChar string) []model.KeyValue {
+	kvs := make([]model.KeyValue, len(tagsMap))
+	i := 0
+	for k, v := range tagsMap {
+		dKey := strings.Replace(k, deDotChar, ".", -1)
+		tag := model.KeyValue{Key: dKey}
+		// TODO v.(int/64/32) didn't work it is always a float
+		if pInt, err := strconv.ParseInt(fmt.Sprintf("%v", v), 10, 64); err == nil {
+			tag.VType = model.Int64Type
+			tag.VInt64 = pInt
+		} else if val, ok := v.(float64); ok {
+			tag.VType = model.Float64Type
+			tag.VFloat64 = val
+		} else if val, ok := v.(bool); ok {
+			tag.VType = model.BoolType
+			tag.VBool = val
+		} else if val, ok := v.(string); ok {
+			tag.VType = model.StringType
+			tag.VStr = val
+			// it's always as string
+		} else if val, ok := v.([]byte); ok {
+			tag.VType = model.BinaryType
+			tag.VBinary = val
+		}
```

Comment:
In theory, it can also be something else. How about adding a log statement for that case?